device/core/src/device_client.ts

```diff
@@ -171,13 +175,77 @@ export class Client extends InternalClient {
       retryOp.retry((opCallback) => {
         /*Codes_SRS_NODE_DEVICE_CLIENT_16_040: [The `uploadToBlob` method shall call the `_callback` callback with an `Error` object if the upload fails.]*/
         /*Codes_SRS_NODE_DEVICE_CLIENT_16_041: [The `uploadToBlob` method shall call the `_callback` callback no parameters if the upload succeeds.]*/
-        this.blobUploadClient.uploadToBlob(blobName, stream, streamLength, opCallback);
+        this._blobUploadClient.uploadToBlob(blobName, stream, streamLength, opCallback);
       }, (err, result) => {
         safeCallback(_callback, err, result);
       });
     }, callback);
   }
 
+  /**
+   * @description      The `uploadToBlobV2GetStorageBlobSAS` gets the linked storage account SAS Token from IoT Hub
+   *
+   * @param {String}    blobName                The name to use for the blob that will be created with the content of the stream.
+   * @param {Callback}  [callback]              Optional callback to call when the upload is complete.
+   * @returns {Promise<UploadParams> | void}    Promise if no callback function was passed, void otherwise.
+   *
+   * @throws {ReferenceException} If blobName is falsy.
+   */
+  uploadToBlobV2GetStorageBlobSAS(blobName: string, callback: Callback<UploadParams>): void;
```

Comment:
>uploadToBlobV2GetStorageBlobSAS [](start = 2, length = 31)

I don't think the "uploadToBlobV2" prefix adds clarity - in fact, it may even be confusing if we ship a v2 SDK. Also, using acronyms may make the name smaller but not necessarily easier to use.
how about simply "getBlobSharedAccessSignature and notifyBlobUploadComplete?